### PR TITLE
ipaserver/install/krainstance.py: chown after write

### PR DESCRIPTION
**Pull request by fcami**
When fs.protected_regular=1 root cannot open temp files that
are owned by other users read-write.
So move os.chown after write.

Fixes: https://pagure.io/freeipa/issue/7906
Signed-off-by: François Cami <fcami@redhat.com>

### CODE REVIEW

**Comment by tiran** [comment]
May I propose that you use the opportunity and replace the ugly code with a nice solution?

```
        # Generate configuration file
        pent = pwd.getpwnam(self.service_user)
        with tempfile.NamedTemporaryFile('w', delete=False) as f:
            config.write(f)
            os.fchown(f.fileno(), pent.pw_uid, pent.pw_gid)
            cfg_file = f.name
```

**Comment by fcami** [comment]
@tiran thanks for the proposal, I've removed the temp file creation from the top of the method and switched to yours.

**Comment by fcami** [comment]
This fix has to be backported to ipa-4-7 so that f29/f28 containers running on f30 hosts do not fail.

File: ipatests/prci_definitions/temp_commit.yaml [hunk 1]
```diff
@@ -45,14 +45,14 @@ jobs:
...
-  fedora-29/temp_commit:
+  fedora-f29/test_vault:
     requires: [fedora-29/build]
     priority: 50
     job:
       class: RunPytest
       args:
-        build_url: '{fedora-29/build_url}'
-        test_suite: test_integration/test_REPLACEME.py
+        build_url: '{fedora-f29/build_url}'
```
Comment (netoarmando) [inline ipatests/prci_definitions/temp_commit.yaml]: ` fedora-29/build`

Comment (netoarmando) [reply ipatests/prci_definitions/temp_commit.yaml]: PR-CI stuck reporting: 
```
Apr 10 20:27:35 ci-large-16 bash[6762]: ERROR: __main__: 'NoneType' object is not subscriptable
```

Comment (fcami) [reply ipatests/prci_definitions/temp_commit.yaml]: Thanks @netoarmando much appreciated.

**Comment by fcami** [comment]
Link to vault test: http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/8eb3a816-5c17-11e9-bd14-fa163ee50fe4/
Removing temp commit.

**Comment by fcami** [comment]
master:

* cf42dc1f2930cba3fca144ad4b7c1c01e9ed9163 ipaserver/install/krainstance.py: chown after write

**Pull request closed without merging**